spec/app/domain/authentication/authn-gce/decoded_token_spec.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'Authentication::AuthnGce::DecodedToken' do
+
+  let(:sub_value) { "some_sub_value" }
+  let(:aud_value) { "some_aud_value" }
+  let(:email_value) { "some_email_value" }
+  let(:instance_name_value) { "some_instance_name_value" }
+  let(:project_id_value) { "some_project_id_value" }
+
+  def decoded_token_hash(token_str)
+    JSON.parse(token_str).to_hash
+  end
+
+  let(:decoded_token_hash_valid) do
+    decoded_token_hash(
+      "{" \
+        "\"aud\": \"#{sub_value}\", " \
+        "\"sub\": \"#{sub_value}\", " \
+        "\"email\": \"#{email_value}\", " \
+        "\"google\": { " \
+          "\"compute_engine\": {" \
+            "\"instance_name\": \"#{instance_name_value}\", " \
+            "\"project_id\": \"#{project_id_value}\"" \
+          "}" \
+        "}" \
+      "}"
+    )
```

Comment:
Fly by comment: For all of these examples, you can avoid the explosion of `"` characters by using either a heredoc like `<<~EOS` or by using `%Q`.

Reply:
but then won't it need to be 
```
%Q({"aud": "#{sub_value}","sub": "#{sub_value}", "email": "#{email_value}","google": {"compute_engine": {"instance_name": "#{instance_name_value}","project_id": "#{project_id_value}"}}})
``` 

to avoid printing `\n`?

Reply:
same goes for `EOS`. 
this:
```
<<-EOS
{
  "aud": "#{sub_value}",
  "sub": "#{sub_value}",
  "email": "#{email_value}",
  "google": {
    "compute_engine": {
      "instance_name": "#{instance_name_value}",
      "project_id": "#{project_id_value}"
    }
  }
}
EOS
```

outputs this:
```
"{\n  \"aud\": \"sub\",\n  \"sub\": \"sub\",\n  \"email\": \"email\",\n  \"google\": {\n    \"compute_engine\": {\n      \"instance_name\": \"ins\",\n      \"project_id\": \"proj\"\n    }\n  }\n}\n"
```

Reply:
If you want it without newlines you can just remove them.


```ruby
sub_value = 99
email_value = 'a@b.com'
instance_name_value = 'foo'
project_id_value = 100

x = <<~EOS
{
  "aud": "#{sub_value}",
  "sub": "#{sub_value}",
  "email": "#{email_value}",
  "google": {
    "compute_engine": {
      "instance_name": "#{instance_name_value}",
      "project_id": "#{project_id_value}"
    }
  }
}
EOS

puts x.gsub("\n", '')

```

[Try it online!][TIO-kebejxbs]

[Ruby]: https://www.ruby-lang.org/
[TIO-kebejxbs]: https://tio.run/##bY/BC4IwFMbv768Y67ACETsaBl06d@gajKlLDN1EXRiy/vW1udCKDoP33u/te9/XqvRhTKdSemeV4miP4hh4zcpqHhB2SMNM1gRK0fVMZJwKVvOFX6Uk0LTyxrOelvkMtlEEMNgiSZ7H0xlGQAgzleMdwqtxvqlx4IDt/4PJjUcfxt6wkLKouKVO3PbWZ6N6TrkoSrHMLfky7@X@5PGy04clkd/@TajxtKnBPQ0uIdjbHRrCwmZY44vAASJkA8a8AA "Ruby – Try It Online"

Reply:
thanks!

btw, i'm a [`replit`](https://repl.it/repls/ElaborateSteelblueLight) fan :)

Reply:
works perfectly, thanks

Reply:
Yeah, I like TIO for sharing snippets because all the code is encoded in the URL itself.  But mainly it's used for code golf because it supports every esoteric language under the sun.  I don't use to actually code ruby with :).